### DISCUSSION TITLE
Aircraft Symbols when Changing Model ICAO

### DISCUSSION THREAD

**Discussion by gonzalu** [vradarserver/vrs#33] [Q&A]
Hello,

Is it possible to have a way to show proper symbols on the map display for aircraft that I have changed the Model ICAO for to represent a flag/silhouette in the LIST view? 

Perhaps leveraging one of the USER fields in BaseStation.sqb? 

Currently the workaround is to use custom css/js etc. which is very dauting for me and frankly confusing and not working very well.  

Currently, for example, I change all DELTA and other major airlines Model ICAO field to represent a specific type or special c/s.

DALA320
AALB738
UALE170 

Maybe instead, in order to keep the MAP display properly showing the correct species, VRS could look at the USER1 field or similar to see if it should use a different file for the LIST display., 

etc. 

Thank you for your consideration.

**Comment by vradarserver** [comment]
VRS only supports standard ICAO 8643 codes in the `ICAOTypeCode` field. There is a separate field, the `OperatorFlagCode`, for operator flags. There is no support (and there never will be support) for mixing two distinct values into a single database field. BaseStation.sqb already plays fast and loose with database normalisation rules, that would be taking things a step too far.

I could add a custom script to the [vrs-custom-scripts](https://github.com/vradarserver/vrs-custom-scripts) repository that uses those two fields to look for <TYPE><OPERATOR> when looking for silhouettes if you like? E.G. if you have "DAL" for the operator and "A320" for the model type then it will look for DALA320.bmp, then A320.bmp and then <Aircraft's ICAO>.bmp for the silhouette.

**Reply by gonzalu** [reply]
Hi Andrew, 

Thanks for the prompt response. That is EXACTLY what my point was. I have been using the filed my way and would like to get it BACK TO default and standard. 

I think (without knowing all the ins and outs) your method will work. Today it is VERY easy for me and it works 100% without any issues and it is rock solid. Whatever I enter into the ICAOTypeCode field, VRS will display that file. The same goes for the Operator ICAO field. 

Now comes the problem for me and likely others. What happens with airlines like JBU/B6? I have different Tail c/s for all models and then the offshoot full custom c/s frames. How would the system handle those? 

For example, B6/JBU. They have a bunch of planes that are cookie cutter A320 s but they each have (or a bunch have) a different tail than another bunch. Today it is easy for me to just enter "JBUA320BLU" in the Model ICAO field identifying the Blueberries tail design and "JBUA320TAR" for the Tartan Tail design. In the Operator Field I enter the same or similar "JBUA320Blueberries" or "JBUA320Tartan" with the obvious length limits of the two fields in mind. 

Again, right now I was looking for a way to correct the displayed species on the map :) as I am happy how I have hacked the database to display the Side View and Logo as I want... it all looks great other than generic icon for all planes on the map display. 

Looking forward to your thoughts on this. If Custom Content can solve my problem great. So far all the workarounds I have used are either very complicated or break other things .. 

If you can think up a way for us to have this flexibility it would be tremendous. 

THANK YOU so much for the work on this addictive software :D

**Reply by gonzalu** [reply]
I'd like to expand on what my original idea was for an enhancement to VRS:

Say, VRS would have an option in settings that reads: 

```
List View Options
      Silhouette
           [x] Use ICAOTypeCode
           [  ] Use UserString1

      Operator Flag
           [x] Use OperatorFlagCode
           [  ] Use UserString2
```

then VRS could let the user decide to use whatever is entered in the fields above or just the defaults 

A mod to the Database Editor Plugin to allow end users to edit those fields would be welcome as well. But II can handle editing the BaseStation file directly in an SQLite editor if need be. 

Thank you...
